src/meili/match_route.cc

```diff
@@ -208,69 +216,152 @@ bool MergeRoute(std::vector<EdgeSegment>& route, const State& source, const Stat
     throw std::logic_error("The first edge must be an origin (invalid predecessor)");
   }
 
-  MergeEdgeSegments(route, segments.rbegin(), segments.rend());
+  // TODO: why doesnt routing.cc return trivial routes? move this logic there
+  // Might be a trivial route
+  if (segments.empty()) {
+    return false; // this is a discontinuity for now
+    // If we have no chance of making a trivial route bail
+    /*const auto& source_edge = source.candidate().edges.front();
+    const auto& target_edge = target.candidate().edges.front();
+    if (source_edge.id != target_edge.id || source_edge.percent_along != target_edge.percent_along) {
+      return false;
+    }
+    // Make a trivial route
+    segments.emplace_back(source_edge.id, source_edge.percent_along, source_edge.percent_along);*/
+  }
+
+  route.insert(route.end(), segments.crbegin(), segments.crend());
   return true;
 }
 
-std::vector<EdgeSegment> MergeRoute(const State& source, const State& target) {
-  std::vector<EdgeSegment> route;
-  MergeRoute(route, source, target);
-  return route;
+void cut_segments(const std::vector<MatchResult>& match_results,
+                  int first_idx,
+                  int last_idx,
```

Comment:
What do these indices correspond to?

Reply:
the subsection of the match results that correspond to this group of edge segments. the match results have all the tracepoints from the users input. we use these indices to focus this function on just those match results that were used to get these edge segments (partial route from one state to the next but not all states).